builder.go

```diff
@@ -361,33 +358,51 @@ type FuncOption struct {
 	key     string
 }
 
-func Option(key string, operand any) FuncOption { return FuncOption{key: key, operand: operand} }
+func VarOption(name, varName string) FuncOption {
+	return FuncOption{key: name, operand: variable(varName)}
```

Comment:
same 
add checking if variable is valid. according to [abnf](https://github.com/unicode-org/message-format-wg/blob/7c00820a0462679eba696181c45bfadb43d2eedd/spec/message.abnf#L88-L95)